feat(APP-234): Enhance security by adding input sanitization and CSP reporting

tyhonchik opened the pull request:
# Description

- Added input sanitization to various components using .
- Introduced  and  components for safe HTML rendering.
- Implemented CSP reporting to Sentry for improved security monitoring.
- Updated middleware to include CSP report URI based on environment variables.

## Type of Change

<!--- Please delete the options that are not relevant. -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Developer Checklist:

- [ ] Manually smoke tested the functionality locally
- [ ] Confirmed there are no new warnings or errors in the browser console
- [ ] (For User Stories only) Double-checked that all Acceptance Criteria are satisfied
- [x] Confirmed there are no new warnings on automated tests
- [x] Merged and published any dependent changes in downstream modules
- [x] Selected the correct base branch
- [x] Commented the code in hard-to-understand areas
- [x] Followed the code style guidelines of this project
- [x] Reviewed that the Files Changed in Github’s UI reflect my intended changes
- [ ] Confirmed the pipeline checks are not failing

## Review Checklist:

- [ ] Tested locally that all Acceptance Criteria or Expected Outcomes are satisfied
- [ ] Confirmed that changes follow the code style guidelines of this project

tyhonchik committed:
feat(APP-234): Enhance input sanitization and CSP reporting

- Added `sanitizePlainText` to various input fields across forms to prevent XSS attacks.
- Introduced a new `sanitizeMode` option in `useFormField` for flexible sanitization strategies.
- Updated Content Security Policy (CSP) to include reporting to Sentry with environment-specific configurations.

thekidnamedkd commented on src/shared/components/SafeDocumentParser.tsx:
```diff
@@ -0,0 +1,13 @@
+import { sanitizeHtmlRich } from '@/shared/security/htmlSanitizer';
```
should probably have it's own folder with index export just to follow style guide as a 'components'

tyhonchik replied:
done, thanks
forgot about barrel files (I'm not a fan of them 🙈)

thekidnamedkd replied:
I can see your point... we were hoping to avoid circular dependencies in areas where we do composed objects like Radix such <SomeComponentGrouping.Footer />

thekidnamedkd commented on src/shared/components/SafeHtml.tsx:
```diff
@@ -0,0 +1,15 @@
+import { sanitizeHtmlRich, sanitizeHtmlStrict } from '@/shared/security/htmlSanitizer';
```
same as SafeDocumentParser

tyhonchik replied:
also done

thekidnamedkd commented on src/shared/components/SafeDocumentParser.tsx:
```diff
@@ -0,0 +1,13 @@
+import { DocumentParser } from '@aragon/gov-ui-kit';
+import type React from 'react';
+
+type DocumentParserProps = React.ComponentProps<typeof DocumentParser>;
```
I think we can just use direct type imports here

tyhonchik replied:
rewrote to direct imports

thekidnamedkd commented on src/shared/components/SafeHtml.tsx:
```diff
@@ -0,0 +1,15 @@
+export interface ISafeHtmlProps extends React.HTMLAttributes<HTMLDivElement> {
+    html: string;
+    variant?: 'rich' | 'strict';
+}
```
same thing, I think we can use direct import and let's add JS docs?

tyhonchik replied:
added JSdoc and direct types

thekidnamedkd commented on src/shared/security/htmlSanitizer.ts:
```diff
@@ -0,0 +1,133 @@
+import DOMPurify from 'isomorphic-dompurify';
```
consider barrel index export for this directory and possibly one level up in shared

tyhonchik replied:
done

thekidnamedkd reviewed (commented):
Looks good! Minor style guide comments mostly.

tyhonchik committed:
feat(APP-234): fix PR comments

- Added a new ESLint rule to discourage the use of `dangerouslySetInnerHTML` in favor of the `SafeHtml` component.
- Added barrel-file for security utils.
- Improved documentation for `SafeHtml` component to clarify sanitization variants.

thekidnamedkd reviewed (approved):
Niceeee ✨
